### PR TITLE
add gas station

### PR DESCRIPTION
**Pull request by AdePhil**
### Describe your change:

Added a new algorithm to solve the gas station problem. I added the file `gas_station.py` to the `greedy_methods` directory. I have also added docstrings for the function with an explanation and doctests.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by AdePhil** [comment]
Hi @cclauss, Can you take a look at this. Thanks.

File: greedy_methods/gas_station.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    total_gas = sum(gas)
+    total_cost = sum(cost)
+
+    if total_gas < total_cost:
+        return -1
+
+    start = 0
+    net = 0
+    for i in range(len(gas)):
```
Comment (cclauss) [inline greedy_methods/gas_station.py:54]: I would call either call the parameters `gas_quantities` and `costs` and then...
```suggestion
    for gas_quantitiy, cost in zip(gas_quantities, costs):
```

Better yet, represent the data as a dataclass.

**Comment by cclauss** [comment]
Please let me try something...

File: greedy_methods/gas_station.py (hunk 1 above)
Comment (AdePhil) [reply greedy_methods/gas_station.py]: @cclauss Thanks for the review. Made the updates.

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Any questions about these changes?
https://github.com/TheAlgorithms/Python/pull/9446/commits/02073549e1b96df92d5e9b5156ae13b9055f8ee5

**Review by cclauss — APPROVED** [review]
Nice one!  Thanks for your patience.

**Comment by AdePhil** [comment]
> Any questions about these changes? [0207354](https://github.com/TheAlgorithms/Python/commit/02073549e1b96df92d5e9b5156ae13b9055f8ee5)
> 
> Tuples are smaller in memory than lists and they cannot be modified at runtime which can be an advantage for data safety. For the doctests, we do not need to test every element but the length and the first and the last is a good idea.

Yeah. This makes sense. Thanks.

**Pull request merged**